src/ResultSet.jl

```diff
@@ -41,45 +41,56 @@ end
 export columns
 
 # accessors 
-columns(rs::ResultSet) = rs.columns
-Base.names(rs::ResultSet) = rs.names
-Base.size(rs::ResultSet) = rs.size
-Base.size(rs::ResultSet, i::Integer) = rs.size[i]
+columns(rs::ResultSet) = getfield(rs, :columns)
+Base.names(rs::ResultSet) = getfield(rs, :names)
+
+Base.size(rs::ResultSet) = getfield(rs, :size)
+Base.size(rs::ResultSet, i::Integer) = getfield(rs, :size)[i]
+Base.length(rs::ResultSet) = getfield(rs, :size)[1]
 
 # Size displayed as a string 
 sizestr(rs::ResultSet) = string(size(rs, 1)) * " rows x " * string(size(rs, 2)) * " columns"
 
 # find index for the column symbol
 function symindex(rs::ResultSet, s::Symbol) 
-    n = findfirst(x -> x == s, rs.names)
+    n = findfirst(x -> x == s, names(rs))
     n == 0 && error("column symbol not found: $s")
     n
 end
 
 # Direct cell access
-Base.getindex(rs::ResultSet, i::Integer, j::Integer) = rs.columns[j][i]
-Base.getindex(rs::ResultSet, i::Integer, s::Symbol) = rs.columns[symindex(rs, s)][i]
-Base.setindex!(rs::ResultSet, val, i::Integer, j::Integer) = rs.columns[j][i] = val
-Base.setindex!(rs::ResultSet, val, i::Integer, s::Symbol) = rs.columns[symindex(rs, s)][i] = val
+Base.getindex(rs::ResultSet, i::Integer, j::Integer) = columns(rs)[j][i]
+Base.getindex(rs::ResultSet, i::Integer, s::Symbol) = columns(rs)[symindex(rs, s)][i]
+Base.setindex!(rs::ResultSet, val, i::Integer, j::Integer) = columns(rs)[j][i] = val
+Base.setindex!(rs::ResultSet, val, i::Integer, s::Symbol) = columns(rs)[symindex(rs, s)][i] = val
 
-# Return a single row as a tuple
-Base.getindex(rs::ResultSet, i::Integer) = Tuple([c[i] for c in rs.columns])
+# Return a single row as a named tuple
+Base.getindex(rs::ResultSet, i::Integer) = 
```

Comment:
One pattern a lot of table types have moved to is having a "lazy row" struct instead of materializing full NamedTuples (which can be extremely costly for really wide datasets, like >1000 columns). It would look something like:
```julia
struct ResultSetRow <: AbstractVector{Any}
    r::ResultSet
    row::Int
end
```
and then you'd define `getindex`, `getproperty`, `size`, and `propertynames` on `ResultSetRow`.

Just something to consider.